packages/grid/_modules_/grid/hooks/root/useApi.ts

```diff
@@ -4,15 +4,26 @@ import { GridSubscribeEventOptions } from '../../utils/eventEmitter/GridEventEmi
 import { useLogger } from '../utils/useLogger';
 import { GRID_COMPONENT_ERROR, GRID_UNMOUNT } from '../../constants/eventsConstants';
 import { useGridApiMethod } from './useGridApiMethod';
+import { MuiEvent } from '../../models/gridOptions';
+
+const isSynthenticEvent = (event: any): event is React.SyntheticEvent => {
+  return event.isPropagationStopped !== undefined;
+};
 
 export function useApi(apiRef: GridApiRef): void {
   const logger = useLogger('useApi');
 
   const publishEvent = React.useCallback(
-    (name: string, params: any, event?: React.SyntheticEvent) => {
-      if (!event || !event.isPropagationStopped || !event.isPropagationStopped()) {
-        apiRef.current.emit(name, params, event);
+    (
+      name: string,
+      params: any,
+      event: MuiEvent<React.SyntheticEvent | DocumentEventMap[keyof DocumentEventMap] | {}> = {},
+    ) => {
+      event.defaultMuiPrevented = false;
```

Comment:
if we set `event.defaultMuiPrevented = true;` and then publish then it will be reset here and be published.
Why not check `defaultMuiPrevented` everywhere there `isPropagationStopped`

Reply:
We still want the event to be published. Only the listeners should not run if `event.defaultMuiPrevented` is `true`.